app/src/main/java/io/homeassistant/companion/android/notifications/NotificationsActivity.kt

```diff
@@ -0,0 +1,227 @@
+package io.homeassistant.companion.android.notifications
+
+import android.content.ActivityNotFoundException
+import android.content.Intent
+import android.database.Cursor
+import android.net.Uri
+import android.os.Bundle
+import android.view.ActionMode
+import android.view.Menu
+import android.view.MenuItem
+import android.view.View
+import android.widget.AbsListView
+import android.widget.AdapterView
+import android.widget.ListView
+import androidx.appcompat.app.AppCompatActivity
+import androidx.cursoradapter.widget.SimpleCursorAdapter
+import io.homeassistant.companion.android.R
+import io.homeassistant.companion.android.settings.SettingsActivity
+import kotlinx.android.synthetic.main.activity_notifications.*
+
+class NotificationsActivity : AppCompatActivity() {
+
+    private var db: NotificationsDB? = null
+    private lateinit var stream: ListView
+    private lateinit var adapter: SimpleCursorAdapter
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_notifications)
+        setSupportActionBar(findViewById(R.id.toolbar))
+        supportActionBar?.setDisplayHomeAsUpEnabled(true)
+
+        loadDatabase()
+        generateStream()
+    }
+
+    override fun onDestroy() {
+
+        super.onDestroy()
+        db?.close()
+    }
+
+    private fun loadDatabase() {
+
+        db = NotificationsDB(this)
+
+        try {
+
+            db?.open()
+        } catch (e: Exception) {
+
+            e.printStackTrace()
+        }
+    }
+
+    private fun generateStream() {
+
+        stream = message_stream
```

Comment:
It might not always be a listview, but it will always be a message stream. Depending on what features we end up giving it it might end up better being a recyclerview, but for initial simplicity I went with listview and simple cursor.